src/input/type_base.hh

```diff
@@ -162,6 +168,14 @@ public:
     /// Print JSON output of attributes to @p stream.
     void write_attributes(ostream& stream) const;
 
+    /// Create instance of generic type
+    virtual boost::shared_ptr<TypeBase> make_instance(std::vector<ParameterPair> vec = std::vector<ParameterPair>()) const
```

Comment:
TypeBase is abstract class anyway, so just make this method abstract as well:
make_instance(...) = 0;